fintalk_world/static/js/base.js

```diff
@@ -33,9 +34,137 @@ const run_app = () => {
                 rightArrow.setAttribute('disabled', '');
             }
         });
+
     };
     indexSlider();
     
+    const apiGenerator = () => {
+        //Api of stokes
+
+        const sp = document.querySelector('#sp'), 
+            dji = document.querySelector('#dji'),
+            ndaq = document.querySelector('#ndaq'), 
+            rsl = document.querySelector('#rsl'),
+            btc = document.querySelector('#btc'), 
+            euUsd = document.querySelector('#eu-usd'), 
+            gbpUsd = document.querySelector('#gbp-usd'), 
+            usdJp = document.querySelector('#usd-jp');
+
+        let sp_symbol = "GSPC",
+            dji_symbol = "DJI",
+            // ftse_symbol = "UKX",
+            nasdaq_symbol = "NDAQ",
+            rut_symbol = "RUT";
+            // nkk_symbol = "NI225",
+            // bond_symbol = "TNX",
+            // crude_symbol = "CL=F",
+            // gold_symbol = "GC=F";
+                   
+            const api_sp = `https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${sp_symbol}&apikey=PNA7JS5UMRI0PG2D`;
+            const api_dji = `https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${dji_symbol}&apikey=PNA7JS5UMRI0PG2D`;
+            const api_nasdaq = `https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${nasdaq_symbol}&apikey=PNA7JS5UMRI0PG2D`;
+            const api_rut = `https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${rut_symbol}&apikey=PNA7JS5UMRI0PG2D`;
+            const api_btc = `https://www.alphavantage.co/query?function=CURRENCY_EXCHANGE_RATE&from_currency=BTC&to_currency=USD&apikey=PNA7JS5UMRI0PG2D`;
+            const api_eu_usd = `https://www.alphavantage.co/query?function=CURRENCY_EXCHANGE_RATE&from_currency=EUR&to_currency=USD&apikey=PNA7JS5UMRI0PG2D`;
+            const api_gbp_usd = `https://www.alphavantage.co/query?function=CURRENCY_EXCHANGE_RATE&from_currency=GBP&to_currency=USD&apikey=PNA7JS5UMRI0PG2D`;
+            const api_usd_jpy = `https://www.alphavantage.co/query?function=CURRENCY_EXCHANGE_RATE&from_currency=USD&to_currency=JPY&apikey=PNA7JS5UMRI0PG2D`;
```

Comment:
Switch all of these to use a `BASE_URL` or a function.
```js
API_KEY = `PNA7JS5UMRI0PG2D`;
BASE_URL = `https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol={symbol}&api_key={API_KEY}`;
```
Then pass in the `symbol` you need using maybe an arrow func?
```js
const getUrl = symbol => `https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol={symbol}&api_key={API_KEY}`;